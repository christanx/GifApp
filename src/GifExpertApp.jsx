import { useState } from 'react'
import AddCategories from './components/AddCategories';
import CategoryGrid from './components/CategoryGrid';

const GifExpertApp = ({ defaultCategories = ['Spiderman'] }) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategories setCategories={setCategories} />
            <hr />
            <ul>
            {
                categories.map(category => (
                    <CategoryGrid key={category} category={category}/>
                ))
            }
            </ul>
            
        </>
    )
}

export default GifExpertApp
