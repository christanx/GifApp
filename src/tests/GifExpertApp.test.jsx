import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"


describe('Testing <GifExpertApp />', () => {

    test('Component successful', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar las categorias en la grid', () => {
        const categories = ['goku', 'broly'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
