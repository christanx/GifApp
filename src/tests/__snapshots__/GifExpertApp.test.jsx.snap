// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> Component successful 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <CategoryGrid
      category="Spiderman"
      key="Spiderman"
    />
  </ul>
</Fragment>
`;

exports[`Testing <GifExpertApp /> Debe mostrar las categorias en la grid 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <CategoryGrid
      category="goku"
      key="goku"
    />
    <CategoryGrid
      category="broly"
      key="broly"
    />
  </ul>
</Fragment>
`;
